### ISSUE TITLE
Showing warning "This  transaction may be risky" and got failed

### ISSUE THREAD

**Issue by AayushOjha** [nitish24p/react-native-upi#32]
In many mobiles I am getting this issue with google-pay and phone-pe that
      'this transaction may be risky. for your safety it cannot be completed at this time'

what could be the possible solution?

**Comment by ajitmaurya4342** [comment]
Same error i am also facing

**Comment by SarathKannan** [comment]
Got similar error...


![20210824_152003.jpg](https://user-images.githubusercontent.com/35830880/130596247-0e2ecd08-6922-4895-97b4-2e35dbba5a34.jpg)

**Comment by 305425** [comment]
same i am facing

**Comment by ImPDP** [comment]
You need to use merchant UPI ID (registered in Business UPI App).

**Comment by karankarakoti** [comment]
any solution? got the same error while using business id

**Comment by nitish24p** [comment]
I dont think this is related to the library persay. You might have to check if this is a valid upi merchant id

**Comment by avinash-capsitech** [comment]
you can solve this error from below
1. transactionRef: 'send every time new transactionRef'.
2. also You need to use merchant UPI ID (registered in Business UPI App).

**Comment by valisettisuresh** [comment]
same 

> Got similar error...
> 
> ![20210824_152003.jpg](https://user-images.githubusercontent.com/35830880/130596247-0e2ecd08-6922-4895-97b4-2e35dbba5a34.jpg)

same problem

**Comment by Dk2626** [comment]
> 

merchant id in the merchant code property, Am I right?

**Comment by sk02769** [comment]
Getting same error

**Comment by sk02769** [comment]
> you can solve this error from below
> 
> 1. transactionRef: 'send every time new transactionRef'.
> 2. also You need to use merchant UPI ID (registered in Business UPI App).
>    It wokred for me !!

Can you show little things how you did it

**Comment by Dk2626** [comment]
Hi, thanks for your reply, I used new transactionRef, but I am getting
"Transaction is risky", I have attached my code!

**Comment by Dk2626** [comment]
![upi payment](https://user-images.githubusercontent.com/52568067/202858898-ef5acff5-49da-4549-bfb6-a7ca93652369.png)
Hi, thanks for your reply, I used new transactionRef, but I am getting
"Transaction is risky", I have attached my code!

**Comment by Dk2626** [comment]
> > you can solve this error from below
> > 
> > 1. transactionRef: 'send every time new transactionRef'.
> > 2. also You need to use merchant UPI ID (registered in Business UPI App).
> >    It wokred for me !!
> 
> Can you show little things how you did it

![upi payment](https://user-images.githubusercontent.com/52568067/202858969-a64860f6-2152-4526-a250-5bfcffc98fc0.png)
Hi, thanks for your reply, I used new transactionRef, but I am getting
"Transaction is risky", I have attached my code!

**Comment by sk02769** [comment]
You are asking or this code worked for you

**Comment by Dk2626** [comment]
It is not worked for me, I am asking

**Comment by sk02769** [comment]
I thought you have done it
have you another way to integrate it

**Comment by sk02769** [comment]
or anythoer npm things

**Comment by avinash-capsitech** [comment]
try this Code,May be it help's you 
 function upiPayment() {
RNUpiPayment.initializePayment(
      {
        vpa: 'Enter your merchantID', // Like:--> 9876543210@upi
        payeeName: 'upi registered', //your shop name which is registered in bank
        amount: getAmount,
        transactionRef:
          isVerification === 'verified' ? transRef : props.transRef, //this check for verification transactionRef(same or new transactionRef)  
      },
      successCallback,
      failureCallback,
    );
  }
  
  function successCallback(data: any) {
    console.log('Succes data :----->', data);
    /* This callback is not returning any type of data so we do 
   jugad in failure mehtod*/
  }
  //worked 
  function failureCallback(data: any) {
    if (data.Status === 'SUCCESS' || data.Status === 'Success') {
            paymentAction();
    } else if (data.Status === 'FAILURE') {
      paymentAction();
    } else {
            props.onDismiss(data.Status);
    }
  }

**Comment by valisettisuresh** [comment]
what is paymentAction()

**Comment by avinash-capsitech** [comment]
> what is paymentAction()

When payment is completed successfully then we call post payment API in which we send success data to server.
Note:- If you don't want anything after successful completion of payment please ignore it.
